File: src/protocol/voice-events.ts
export const mobileVoiceEventTypes = [
  "voice_ack",
  "voice_state",
  "voice_segment",
  "voice_triggers",
  "voice_cue",
  "voice_assistant_text",
  "voice_speak_request",
  "voice_tts_unavailable",
  "voice_cancel_speech",
  "voice_subagent_started",
  "voice_subagent_progress",
  "voice_subagent_report",
  "voice_subagent_failed",
  "voice_warning",
  "summary",
  "error",
] as const;

export type MobileVoiceEventType = (typeof mobileVoiceEventTypes)[number];

File: apps/ios/NearMind/NearMind/Protocol/VoiceEvents.swift
import Foundation

enum VoiceEventType: String, CaseIterable {
    case voiceAck = "voice_ack"
    case voiceState = "voice_state"
    case voiceSegment = "voice_segment"
    case voiceTriggers = "voice_triggers"
    case voiceCue = "voice_cue"
    case voiceAssistantText = "voice_assistant_text"
    case voiceSpeakRequest = "voice_speak_request"
    case voiceCancelSpeech = "voice_cancel_speech"
    case voiceSubagentStarted = "voice_subagent_started"
    case voiceSubagentProgress = "voice_subagent_progress"
    case voiceSubagentReport = "voice_subagent_report"
    case voiceSubagentFailed = "voice_subagent_failed"
}
